src/repr/src/adt/datetime.rs

```diff
@@ -1199,9 +1196,15 @@ fn fill_pdt_date(
     // Check for one number that represents YYYYMMDDD.
     match actual.front() {
         Some(Num(mut val, digits)) if 6 <= *digits && *digits <= 8 => {
-            pdt.day = Some(DateTimeFieldValue::new(val % 100, 0));
+            pdt.day = Some(DateTimeFieldValue::new(
```

Comment:
These `unwraps` are a bit scary. Are we sure that it will never fail? It may just be safer to return errors here instead of unwrapping.

Reply:
Thanks for the comments! I have pushed a commit trying to address them now. Moritz gave me some feedback of when to use `expect` vs. `map_err`, so I tried to already use the appropriate construct depending on the situation. Just let me know if any choice between the two looks odd!